RecoBTag/SecondaryVertex/plugins/TemplatedSecondaryVertexProducer.cc

```diff
@@ -1075,6 +1148,60 @@ void TemplatedSecondaryVertexProducer<IPTI,VTX>::matchSubjets(const std::vector<
    }
 }
 
+// ------------ method that matches subjets and original fat jets ------------
+template<class IPTI,class VTX>
+void TemplatedSecondaryVertexProducer<IPTI,VTX>::matchSubjets(const edm::Handle<edm::View<reco::Jet> >& fatJets,
+                                                              const edm::Handle<std::vector<IPTI> >& subjets,
+                                                              std::vector<std::vector<int> >& matchedIndices)
+{
+   for(size_t fj=0; fj<fatJets->size(); ++fj)
+   {
+     std::vector<int> subjetIndices;
+     size_t nSubjetCollections = 0;
+     size_t nSubjets = 0;
+
+     const pat::Jet * fatJet = dynamic_cast<const pat::Jet *>( fatJets->ptrAt(fj).get() );
+
+     if( !fatJet )
+       edm::LogError("WrongJetType") << "Wrong jet type for input fat jets. Please check that the input fat jets are of the pat::Jet type.";
+     else
+       nSubjetCollections = fatJet->subjetCollectionNames().size();
+
+     if( nSubjetCollections>0 )
+     {
+       for(size_t coll=0; coll<nSubjetCollections; ++coll)
+       {
+         const pat::JetPtrCollection & fatJetSubjets = fatJet->subjets(coll);
+
+         for(size_t fjsj=0; fjsj<fatJetSubjets.size(); ++fjsj)
+         {
+           ++nSubjets;
+
+           for(size_t sj=0; sj<subjets->size(); ++sj)
+           {
+             const pat::Jet * subJet = dynamic_cast<const pat::Jet *>( subjets->at(sj).jet().get() );
+             if( !subJet )
+               edm::LogError("WrongJetType") << "Wrong jet type for input subjets. Please check that the input subjets are of the pat::Jet type.";
```

Comment:
After this line, there needs to be a `break;` or `else if...`; otherwise there could be a segmentation fault.